opensearch/src/main/java/org/opensearch/sql/opensearch/storage/script/aggregation/AggregationQueryBuilder.java

```diff
@@ -112,10 +112,20 @@ public Map<String, OpenSearchDataType> buildTypeMapping(
       List<NamedAggregator> namedAggregatorList,
       List<NamedExpression> groupByList) {
     ImmutableMap.Builder<String, OpenSearchDataType> builder = new ImmutableMap.Builder<>();
-    namedAggregatorList.forEach(agg -> builder.put(agg.getName(),
-        OpenSearchDataType.of(agg.type())));
-    groupByList.forEach(group -> builder.put(group.getNameOrAlias(),
-        OpenSearchDataType.of(group.type())));
+    namedAggregatorList.forEach(agg -> {
+      if (OpenSearchDateType.isDateTypeCompatible(agg.type())) {
+        builder.put(agg.getName(), OpenSearchDateType.of(agg.type()));
+      } else {
+        builder.put(agg.getName(), OpenSearchDataType.of(agg.type()));
+      }
+    });
+    groupByList.forEach(group -> {
+      if (OpenSearchDateType.isDateTypeCompatible(group.type())) {
+        builder.put(group.getNameOrAlias(), OpenSearchDateType.of(group.type()));
```

Comment:
Move this logic into `OpenSearchDataType.of`, so it will check compatibility and return the right object.

Reply:
Done in https://github.com/Bit-Quill/opensearch-project-sql/pull/268/commits/58a8f193ffe1407665734d55a60e53ff9bc97aa8